app/models/embed/purl_json_loader.rb

```diff
@@ -49,6 +51,17 @@ def embargoed
       json.dig('access', 'embargo').present?
     end
 
+    def location_restriction
```

Comment:
This works too but am wondering about possible connections with the resource_file/file implementations: https://github.com/sul-dlss/sul-embed/blob/main/app/models/embed/purl/resource_file.rb#L24 and https://github.com/sul-dlss/sul-embed/blob/main/app/models/embed/purl/file_json_deserializer.rb#L23, although for file, it only seems to be checking for download whereas you are checking for both view and download.

Reply:
ResourceFile is the files on the Purl. Basically in order to do what you are suggesting I would need to go into the structural and look at every file to get this data, which is already in the access field in Cocina.